### ISSUE TITLE
Odd behaviour when dynamically changing columns

### ISSUE THREAD

**Issue by georgecheyne** [angular-ui/ui-grid#6501]
Hi,

I've put together a plunkr to demonstrate the issue I'm seeing:

https://plnkr.co/edit/vxMbbtWMuwJtkGrQqSXR?p=preview

As you click the buttons to switch between years the grid appears to be using a cached colDef for the columns Jan, Feb and Mar.  The cache seems to be keyed on the name of the column rather than the field so the data doesn't show.

This behaviour may be by design?

Obviously the problem can be solved in this case by adding the year to the column name (this is the workaround I have in place).  If this is something you choose to patch it would be nice for my usecase as it's a bit redundant to have the year in the column name when a user has selected it.

Probably pretty low priority but I thought I'd point it out in case.

I couldn't find any other issues that seemed similar but it's quite a hard case to search for.

Thanks,

George

**Comment by mportuga** [comment]
I did a little research into your issue and found that it was actually user error. If you look at the description for the "name" attribute in the [API documentation](http://ui-grid.info/docs/#!/api/ui.grid.class:GridColumn) it states that "This must be unique to each column on a web page since it can be used as a key for retrieving information such as custom sort algorithms."

Based on that I believe that what you actually wanted to use instead of "name" was in fact "displayName". I actually tested that by forking your plunker and my assumption proved to be correct.
https://plnkr.co/edit/vO4qYfSLc9WUfIeaU8bL?p=preview
